### ISSUE TITLE
missing provider-specific pool in visualization

### ISSUE THREAD

**Issue by carns** [mochi-hpc/mochi-json-vis#8]
![foo](https://user-images.githubusercontent.com/5541937/174900297-3519ce1b-f222-4f9a-86aa-6f4267e4d72f.png)
[foo.json.gz](https://github.com/mochi-hpc/mochi-json-vis/files/8952906/foo.json.gz)

See attached json example and png.  For some reason the quintain_rpcs pool is not showing up, but I don't know if my json is mis-formatted or if this is a condition that mochi-json-vis doesn't handle.

I also tried removing the enclosing object, but I got the same result.  This was emitted in a mochi-quintain output file, not bedrock-query, so I'm not sure what the outer-most syntax is supposed to look like.

**Comment by roblatham00** [comment]
thanks for the reproducer.  When i feed this to bedrock I get ` parse error at line 1, column 21: syntax error while parsing value - unexpected ':'; expected end of input`

When i remove the outer-most syntax (first line is now just `{` )  I get 
```
] Could not find pool "" in SSG configuration
```

I pointed SSG to the `__primary__` pool.

with those two changes, Is this now the configuration you were expecting to see?

![quintain](https://user-images.githubusercontent.com/2278417/175108965-94d62d4b-8b8f-484e-a9da-4c91247d1161.png)

Still a good test case to handle and I'm not closing: just want to validate the solution.

**Comment by roblatham00** [comment]
Additional data:  when I grab the json from bedrock, back to the original defective image.

**Comment by mdorier** [comment]
@carns your json looks weird, it starts with `"quintain-provider" : `; this is not a valid JSON...

**Comment by carns** [comment]
> 

Yep, that's what I was expecting to see.

**Comment by carns** [comment]
> @carns your json looks weird, it starts with `"quintain-provider" : `; this is not a valid JSON...

Right, sorry about that.  You can chop that part off, that is an artifact of the quintain output, not the way bedrock natively presents it.

**Comment by roblatham00** [comment]
Well, I have *a* fix but I don't know how great it is.   nlohman-json happily parses the input even with the `"quintain-provider" :`  part.  It just considers the entire input to be a `string` type.

```
% ../build/src/json-viz quintain.json
input is not a JSON object.  Input is string
```

considering my target audience (people like me who are new to JSON), that's not a great error message.    Open to suggestions...

**Comment by carns** [comment]
Oh, I wouldn't worry about handling the malformed json that I gave you.  If it can just draw it with the `"quintain-provider" : ` part lopped off, that would be fine :)

**Comment by roblatham00** [comment]
I can special case that easy enough... "consume from quintain" is a good use case

**Comment by roblatham00** [comment]
OK, I think https://github.com/mochi-hpc/mochi-json-vis/compare/dev-malformed-json does the trick

**Comment by carns** [comment]
This still doesn't work for me.  If I leave the `quintain-provider:` part on the beginning I get this:

```
[json.exception.parse_error.101] parse error at line 1, column 20: syntax error while parsing value - unexpected ':'; expected end of input
```
If I chop that off (I don't really mind that, I ought to just fix quintain to not emit broken syntax) then json-vis doesn't throw any errors, but the resulting dot won't render:

```
Error: <stdin>: syntax error in line 20 near '-'
```

That line looks like it is missing a destination?
```
    quintain-ssg -> ;
```
